### PR TITLE
[Indexing] If module aliasing is used, store module real names to index files

### PR DESCRIPTION
**Pull request by elsh**
If module aliasing is used, store module real names to index unit/record files
Resolves rdar://82896373

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci test

**Review by nathawes — APPROVED** [review]
LGTM!

File: include/swift/AST/USRGeneration.h [hunk 1]
```diff
@@ -47,8 +47,14 @@ bool printDeclTypeUSR(const ValueDecl *D, raw_ostream &OS);
...
 /// Prints out the USR for the given ModuleEntity.
+/// \param useRealNameIfAliased Whether to use the module's real name in case
+///                             module aliasing is used. For example, if a file
+///                             has `import Foo` and `-module-alias Foo=Bar` is
+///                             passed, treat Foo as an alias and Bar as the real
+///                             module name as its dependency. This only applies
+///                             to Swift modules.
 /// \returns true if it failed, false on success.
-bool printModuleUSR(ModuleEntity Mod, raw_ostream &OS);
+bool printModuleUSR(ModuleEntity Mod, raw_ostream &OS, bool useRealNameIfAliased = false);
```
Comment (benlangmuir) [inline include/swift/AST/USRGeneration.h:57]: Is there ever a reason to make a USR using a non-canonical name?  I would have expected USRs to always use the real name unconditionally.  It's not good for us to have two USRs referring to the same thing.   If we really need this to be configurable, the default should probably be flipped, or else there should be no default value.

**Comment by elsh** [comment]
@swift-ci smoke test

**Review by benlangmuir — APPROVED** [review]

**Pull request merged**